mcd-sel-fj/mcd-sel-fj.py

```diff
@@ -53,128 +55,113 @@
-	# Constructing the Dataframe and Exporting it to File  #
-	# ---------------------------------------------------- #
-	
-	product_list_df = pd.DataFrame(product_list)
-	product_list_df.drop_duplicates(
-	    subset=None, keep='last', inplace=True, ignore_index=True)
-	product_list_df.reset_index(drop=True, inplace=True)
-	product_list_df.index = pd.RangeIndex(
-	    start=1, stop=(len(product_list_df.index) + 1), step=1)
-	
-	print(product_list_df)
-	print()
-	
-	timestamp = str(local_datetime.strftime("[%Y-%m-%d %H:%M:%S]"))
-	
-	output_file = str(timestamp + " mcd-sel-fj.csv")
-	output_dir = Path("./scraped-data")
-	
-	# Create directory as required; won't raise an error if directory already exists
-	output_dir.mkdir(parents=True, exist_ok=True)
-	
-	product_list_df.to_csv((output_dir / output_file),
-	                       float_format="%.2f", encoding="utf-8")
-	
-	# Output filename format: "[YYYY-MM-DD hh:mm:ss] mcd-sel-fj.csv"
-	
-	
-	time.sleep(5)
-	
-	browser.quit()
+    # --------------------------------------- #
+    # Getting the Live Exchange Rate from XE  #
+    # --------------------------------------- #
+
+    # Getting the correct XE webpage (all elements)
+    XE = browser.get(
+        "https://www.xe.com/currencyconverter/convert/?Amount=1&From=FJD&To=USD")
+
+    # Scraping the text from the selected element (CSS selector)
+    # findall() and select() methods return a list, indicate index [0] to extract the first element as a string value
+    # Extracting only the number from the text string and converting it to a float value (decimal number)
+    exchange_rate = float(re.findall(
+        r"[-+]?(?:\d*\.\d+|\d+)", browser.find_element(By.CSS_SELECTOR, "p.result__BigRate-sc-1bsijpp-1.iGrAod").text)[0])
+
+    print(exchange_rate)
+    print()
+
+    # -------------------------------------- #
+    # Parsing the data into Dictionary List  #
+    # -------------------------------------- #
+
+    # Vendor updated page URL
+    browser.get("https://fijieats.com/vendor/mcdonalds-nadi")
+
+    time.sleep(6)
+
+    product_list = []
+
+    # Revamped page has menu items split across different html section elements
+    submenus = browser.find_elements(
+        By.CSS_SELECTOR, "section.scrolling_section"
+    )
+
+    # Outer For Loop iterates through each section element
+    for submenu in submenus:
+
+        # Category Names should only contain alphabetical characters
+        category = " ".join(
+            re.findall(
+                r"[a-zA-Z]+",
+                ((submenu.find_element(By.CSS_SELECTOR, "h2")).text)
+            )
+        )
+
+        items = submenu.find_elements(By.CSS_SELECTOR, "div.price_head")
+
+        # Inner For Loop iterates through each div containing a Menu Item
+        for item in items:
+            product = {}
+            product["Date"] = Date
+            product["Day"] = Day
+            product["Territory"] = "Fiji"
+            product["Menu Item"] = (item.find_element(
```

Comment:
Instead, a more conventional double For Loop script (much like the static McD sites) is sufficient.